pkg/autodiscovery/configmgr.go

```diff
@@ -78,216 +78,321 @@ type configManager interface {
 	mapOverLoadedConfigs(func(map[string]integration.Config))
 }
 
-// simpleConfigManager implements the "simple" config manager that reconciles
-// services and templates without any priority, using a store as a backend.
-type simpleConfigManager struct {
+// serviceAndADIDs bundles a service and its associated AD identifiers.
+type serviceAndADIDs struct {
+	svc   listeners.Service
+	adIDs []string
+}
+
+// reconcilingConfigManager implements the a config manager that reconciles
+// services and templates to generate the scheduled configs.
+type reconcilingConfigManager struct {
+	// updates to this data structure work from the top down:
+	//
+	//  1. update orctiveConfigs / activeServices
+	//  2. update templatesByADID or servicesByADID to match
+	//  3. update serviceResolutions, generating changes (see reconcileService)
+	//  4. update scheduledConfigs
+	//
+	// For non-template configs, only steps 1 and 5 are required.
```

Comment:
Where's step 5? :)

Reply:
Ah, I removed a step so this should be "1 and 4".